src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Stream.cs

```diff
@@ -98,22 +98,36 @@ public async Task SendRequestBodyAsync(CancellationToken cancellationToken)
                     {
                         using (Http2WriteStream writeStream = new Http2WriteStream(this))
                         {
-                            await _request.Content.CopyToAsync(writeStream, null, cancellationToken).ConfigureAwait(false);
+                            // TODO: until #9071 is fixed, cancellation on content.CopyToAsync does not work.
+                            // To work around it, register delegate and set _abortException as needed.
+                            using (cancellationToken.Register(stream => { if (((Http2Stream)stream)._abortException == null) ((Http2Stream)stream)._abortException = new OperationCanceledException(); }, this))
```

Comment:
This can use UnsafeRegister instead of Register. Nothing in the callback needs ExecutionContext flowed.

Reply:
You marked this as resolved, but it wasn't addressed.